plansys2_bt_actions/src/plansys2_bt_actions/BTAction.cpp

```diff
@@ -53,19 +53,16 @@ BTAction::on_configure(const rclcpp_lifecycle::State & previous_state)
     RCLCPP_INFO_STREAM(get_logger(), "plugin: [" << plugin << "]");
   }
 
-  BT::BehaviorTreeFactory factory;
   BT::SharedLibrary loader;
 
   for (auto plugin : plugin_lib_names) {
-    factory.registerFromPlugin(loader.getOSName(plugin));
+    factory_.registerFromPlugin(loader.getOSName(plugin));
   }
 
   auto node = rclcpp::Node::make_shared(std::string(get_name()) + "_aux");
   blackboard_ = BT::Blackboard::create();
   blackboard_->set("node", node);
 
-  tree_ = factory.createTreeFromFile(bt_xml_file_, blackboard_);
```

Comment:
moving this down to the `on_activate` function will require moving the ZMQ section just below this to the `on_activate` function as well. you may also have to add `publisher_zmq_.reset()` after the call to `haltTree` in `on_deactivate`

Reply:
opened up a PR to add this change to this branch: #93

Reply:
Ok, I will merge after this

Reply:
@fmrico that PR is set to merge into this branch, not master. I'd recommend merging it into this branch before merging this branch into master.

Reply:
Merged!! :wink: